src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -167,10 +168,29 @@ private static IOperation CreateInternal(BoundNode boundNode)
                 case BoundKind.ExpressionStatement:
                     return CreateBoundExpressionStatementOperation((BoundExpressionStatement)boundNode);
                 default:
-                    return Operation.CreateOperationNone(boundNode.HasErrors, boundNode.Syntax);
+                    var constantValue = ConvertToOptional((boundNode as BoundExpression)?.ConstantValue);
+                    return Operation.CreateOperationNone(boundNode.HasErrors, boundNode.Syntax, constantValue, getChildren: () => GetIOperationChildren(boundNode));
             }
         }
 
+        private static ImmutableArray<IOperation> GetIOperationChildren(BoundNode boundNode)
+        {
+            var boundNodeWithChildren = (IBoundNodeWithIOperationChildren)boundNode;
+            if (boundNodeWithChildren.Children.IsDefaultOrEmpty)
+            {
+                return ImmutableArray<IOperation>.Empty;
+            }
+
+            var builder = ArrayBuilder<IOperation>.GetInstance(boundNodeWithChildren.Children.Length);
```

Comment:
boundNodeWithChilred.Children.SelectAsArray(n => Create(n)); 

might be about same perf?